tests/parsers/setupapi.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""Tests for the Windows Setupapi log parser."""
+
+from __future__ import unicode_literals
+
+import unittest
+
+from plaso.formatters import setupapi as _  # pylint: disable=unused-import
+from plaso.parsers import setupapi
+
+from tests.parsers import test_lib
+
+
+class SetupapiLogUnitTest(test_lib.ParserTestCase):
+  """Tests for the Windows Setupapi log parser.
+
+  Since Setupapi logs record in local time, these tests assume that the local
+  timezone is set to UTC.
+  """
+
+  def testParseDevLog(self):
+    """Tests the Parse function on dev log."""
+    parser = setupapi.SetupapiLogParser()
+    storage_writer = self._ParseFile(['setupapi.dev.log'], parser)
+
+    self.assertEqual(storage_writer.number_of_warnings, 0)
+    self.assertEqual(storage_writer.number_of_events, 194)
+
+    events = list(storage_writer.GetEvents())
+
+    event = events[0]
+
+    self.CheckTimestamp(event.timestamp, '2015-11-22 17:59:28.110000')
+
+    event = events[1]
+
+    self.CheckTimestamp(event.timestamp, '2016-10-05 11:16:03.747000')
+
+    event = events[2]
+
+    self.CheckTimestamp(event.timestamp, '2016-10-05 11:16:16.471000')
+
```

Comment:
Seeing upcoming changes: https://github.com/log2timeline/plaso/pull/2695

to add:
```
event_data = self._GetEventDataOfEvent(storage_writer, event)
```